src/app/users/users.component.ts

```diff
@@ -120,22 +118,30 @@ export class UsersComponent implements OnInit, AfterViewInit {
   }
 
   addRole(user) {
-    // If user has no previous role, add learner role
-    let selectedRolesArray = this.selectedRolesMap.get(user.name);
-    if (this.selectedRolesMap.get(user.name) === undefined || this.selectedRolesMap.get(user.name).length === 0) {
-      selectedRolesArray = [ 'learner' ];
-    }
-    const tempUser = { ...user, roles: [ ...selectedRolesArray ] };
-    this.couchService.put('_users/org.couchdb.user:' + tempUser.name, tempUser).subscribe((response) => {
-      console.log('Success!');
-      this.initializeData();
+    let oldRoles: string[] = [];
+    let selectedRole: string[] = [];
+    this.couchService.get('_users/org.couchdb.user:' + user.name).subscribe((data) => {

```

Comment:
We already have the user information from `initializeData()` stored on the component and passed in as the `user` variable, so we do not need to query the CouchDB before getting it.